ldms/src/core/ldms_rail.c

```diff
@@ -1499,3 +1550,115 @@ ldms_t __ldms_xprt_to_rail(ldms_t x)
 	struct ldms_rail_ep_s *ep = ldms_xprt_ctxt_get(x);
 	return ((ep)?(ldms_t)ep->rail:NULL);
 }
+
+static ldms_set_t __rail_set_by_name(ldms_t _x, const char *set_name)
+{
+	struct ldms_set *set;
+	struct rbn *rbn = NULL;
+	int i;
+	struct ldms_rail_s *r = (void*)_x;
+	struct ldms_xprt *x;
+
+	assert(XTYPE_IS_RAIL(r->xtype));
+
+	__ldms_set_tree_lock();
+	set = __ldms_find_local_set(set_name);
+	__ldms_set_tree_unlock();
+	if (!set)
+		return NULL;
+	for (i = 0; i < r->n_eps; i++) {
+		x = r->eps[i].ep;
+		pthread_mutex_lock(&x->lock);
+		rbn = rbt_find(&x->set_coll, set);
+		pthread_mutex_unlock(&x->lock);
+		if (rbn) {
+			/* found */
+			break;
+		}
+	}
+	if (!rbn) {
+		/* no set found in any of the xprt */
+		ref_put(&set->ref, "__ldms_find_local_set");
+		set = NULL;
+	}
+	return set;
+}
+
+/* defined in ldms_xprt.c */
+size_t format_set_delete_req(struct ldms_request *req, uint64_t xid,
+			     const char *inst_name);
+
+/* Called from ldms_xprt.c.
+ * Tell the peer that have an RBD for this set that it is being
+ * deleted. When they all reply, we can delete the set.
+ */
+void __rail_set_delete(ldms_t _r, struct ldms_set *s,
```

Comment:
It would be great if the function name format differs from the `rails` operation interfaces, e.g., `__rail_listen()`.

Reply:
any suggestion? the '__' was just meant to mark private, in a sense that application should not call it (it is not exported in ldms.h ...).

Reply:
I agree with you that `__rail_on_set_delete` helps at least to me to know that it's that an operation.